File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s=set(nums)
        longest=0

        for x in s:
            if not x-1 in s:
                a=x
                length =1
                while a+1 in s:
                    length+=1
                    a+=1
                longest= max(longest,length)
        return longest  

        
        
        

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        TreeSet<Integer> x = new TreeSet<>();
        for(int i : nums){
            x.add(i);
        }
        int count=1;
        int max=0;
        for(int i :x){
            if(x.contains(i+1))
                count++;
            else
                count=1;
            
            if(max<count)
                max=count;
        }
        return max;
    }
}
